packages/drivers/odsp-driver/src/odspDocumentServiceFactory.ts

```diff
@@ -35,3 +41,62 @@ export class OdspDocumentServiceFactory
         );
     }
 }
+
+/**
+ * Factory for creating sharepoint document service with a provided snapshot.
+ * Use if you don't want to connect to any kind of external/internal storages and want to provide
+ * content directly.
+ */
+export class LocalOdspDocumentServiceFactory extends OdspDocumentServiceFactoryCore {
+    private logger: TelemetryLogger | undefined;
+
+    constructor(
+        private readonly localSnapshot: Uint8Array | string,
+    ) {
+        super(
+            async (_options) => {
+                return this.throwUnsupportedUsage("Getting storage token");
+            },
+            async (_options) => {
+                return this.throwUnsupportedUsage("Getting websocket token");
+            },
+            async () => {
+                return this.throwUnsupportedUsage("Getting SocketIO Client");
+            },
+        );
+    }
+
+    private throwUnsupportedUsage<T>(unsupportedFuncName: string): T {
```

Comment:
It feels like this function should return `never` - see unreachableCase() as example